### PR TITLE
extract sendmail mechanism to config

### PR DESCRIPTION
**Pull request by mabiede**
- adds the possibility to specify sendmail mechanism via configuration ([new colombe release](https://github.com/ocaml/opam-repository/pull/22590))

please release a new version with this feature

### CODE REVIEW

File: lib/sendmail_handler.ml [hunk 1]
```diff
@@ -21,9 +21,11 @@ let rdwr =
...
       (fun { ic; _ } bytes off len ->
         let open Lwt.Infix in
-        let res = Lwt_io.read_into ic bytes off len >>= function
-           | 0 -> Lwt.return `End
-           | len -> Lwt.return (`Len len)
+        let res =
+          Lwt_io.read_into ic bytes off len
+          >>= function
+          | 0 -> Lwt.return `End
+          | len -> Lwt.return (`Len len)
```
Comment (mikonieminen) [inline lib/sendmail_handler.ml:28]: This formatting looks a bit weird to me. I would prefer the old one.

**Review by mikonieminen — APPROVED** [review]
Looks good to me, but formatting looks a bit odd in one place.

File: lib/sendmail_handler.ml (hunk 1 above)
Comment (mabiede) [reply lib/sendmail_handler.ml:28]: @mikonieminen The formatting was changed by the ocamlformat settings.
Adding `break-infix-before-func = false` to .ocamlformat would result in your expected formatting.
Should I add this as well?

Comment (joseferben) [reply lib/sendmail_handler.ml:28]: Imo we should stick to the vanilla ocamlformat config (with the janestreet profile).

**Pull request merged**